File: examples-sdk/csharp/013-basic-nats-worker/Main.cs
// 013-basic-nats-worker — C# SDK equivalent
// Compile: vil compile --from csharp --input 013-basic-nats-worker/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("nats-worker", 8080);
var nats = new ServiceProcess("nats");
nats.Endpoint("GET", "/nats/config", "nats_config");
nats.Endpoint("POST", "/nats/publish", "nats_publish");
nats.Endpoint("GET", "/nats/jetstream", "jetstream_info");
nats.Endpoint("GET", "/nats/kv", "kv_demo");
server.Service(nats);
var root = new ServiceProcess("root");
server.Service(root);
server.Compile();

File: examples-sdk/kotlin/013-basic-nats-worker/main.kt
// 013-basic-nats-worker — Kotlin SDK equivalent
// Compile: vil compile --from kotlin --input 013-basic-nats-worker/main.kt --release

fun main() {
    val server = VilServer("nats-worker", 8080)
    val nats = ServiceProcess("nats")
    nats.endpoint("GET", "/nats/config", "nats_config")
    nats.endpoint("POST", "/nats/publish", "nats_publish")
    nats.endpoint("GET", "/nats/jetstream", "jetstream_info")
    nats.endpoint("GET", "/nats/kv", "kv_demo")
    server.service(nats)
    val root = ServiceProcess("root")
    server.service(root)
    server.compile()
}
